.github/workflows/ci-tests.yml

```diff
@@ -26,7 +26,13 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -e ".[tests]"
+          pip install poetry
```

Comment:
Suggest we change the way that poetry is installed into the CI environment.

Many threads suggests that we'll have issues if poetry is pip-installed in the same environment as a package, for example: https://github.com/python-poetry/poetry/issues/2102

The recommended way to install poetry in a GitHub action seems to be with this action:
https://github.com/snok/install-poetry

This has added bonus of enabling caching of dependencies, which should speed up our CI pipeline significantly.